storage/storage.go

```diff
@@ -137,6 +137,11 @@ func (s *Storage) Read(txn Transaction, path ast.Ref) (interface{}, error) {
 	return doc, nil
 }
 
+// Write updates a value in storage.
+func (s *Storage) Write(txn Transaction, op PatchOp, ref ast.Ref, value interface{}) error {
+	return s.builtin.Write(txn, op, ref, value)
+}
+
```

Comment:
Writes with transactions implies (at least to me) that if we abort the transaction then all the writes done are rolled back, and that the writes are Atomic (e.g. external world sees all the writes or none of them).  I don't see a rollback yet.  Perhaps we're thinking that's something we could add later as the need arises.